### PR TITLE
coprocessor, dag: support decode Time from tipb.Expr

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

File: src/coprocessor/dag/expr/mod.rs [hunk 1]
```diff
@@ -379,6 +380,15 @@ impl Expression {
                 .map(Datum::F64)
                 .map(|e| Expression::new_const(e, tp))
                 .map_err(Error::from),
+            ExprType::MysqlTime => expr.get_val()
+                .decode_u64()
+                .and_then(|ts| {
```
Comment (BusyJay) [inline src/coprocessor/dag/expr/mod.rs]: `ts` is confusing considering this is not a timestamp actually.

File: src/coprocessor/dag/expr/mod.rs [hunk 2]
```diff
@@ -379,6 +380,15 @@ impl Expression {
...
+                    let fsp = expr.get_field_type().get_decimal() as i8;
+                    let tp = expr.get_field_type().get_tp() as u8;
+                    Time::from_packed_u64(ts, tp, fsp, &FixedOffset::east(0))
```
Comment (BusyJay) [inline src/coprocessor/dag/expr/mod.rs]: Should use timezone from context.

File: src/coprocessor/dag/expr/mod.rs [hunk 3]
```diff
@@ -379,6 +380,20 @@ impl Expression {
...
                 .map_err(Error::from),
+            ExprType::MysqlTime => expr.get_val()
+                .decode_u64()
+                .and_then(|i| {
+                    let fsp = expr.get_field_type().get_decimal() as i8;
+                    let tp = expr.get_field_type().get_tp() as u8;
+                    let tz = if tp == types::TIMESTAMP {
+                        ctx.tz
+                    } else {
+                        FixedOffset::east(0)
```
Comment (BusyJay) [inline src/coprocessor/dag/expr/mod.rs]: Why?

Comment (hicqu) [inline src/coprocessor/dag/expr/mod.rs]: According to [TiDB](https://github.com/pingcap/tidb/blob/master/expression/expr_to_pb.go#L118), all timestamp should be adjust to timezone UTC.

Comment (BusyJay) [inline src/coprocessor/dag/expr/mod.rs]: I don't think it's correct. All time should be converted to the tz specified in context.

**Pull request merged**